test/static_arrays.jl

```diff
@@ -17,13 +17,11 @@ for alg in (nothing, LUFactorization(), SVDFactorization(), CholeskyFactorizatio
     sol = solve(LinearProblem(A, b), alg)
     @inferred solve(LinearProblem(A, b), alg)
     @test norm(A * sol .- b) < 1e-10
-
-    if __non_native_static_array_alg(alg)
-        @test_broken __solve_no_alloc(A, b, alg)
+    if alg isa Function && alg === KrylovJL_GMRES
```

Comment:
```suggestion
    if alg isa KrylovJL
```
I tested in the REPL:
```julia
julia> using LinearSolve

julia> alg = KrylovJL_GMRES()
KrylovJL{typeof(Krylov.gmres!), Int64, Tuple{}, Base.Pairs{Symbol, Union{}, Tuple{}, @NamedTuple{}}}(Krylov.gmres!, 0, 0, (), Base.Pairs{Symbol, Union{}, Tuple{}, @NamedTuple{}}())

julia> alg isa KrylovJL
true
```